src/js/utils/date-utils.js

```diff
@@ -0,0 +1,258 @@
+            getDisplayTime,
+            getDisplayLang,
+            getDisplayFormat,
+            isValid;
+
+        var DATE_FORMATS = {
+            DEFAULT_SHORT_DATE_FORMAT: 'll', // example: Jan 01, 2016
+            DEFAULT_LONG_DATE_FORMAT: 'LLLL', // example: Friday, January 01, 2016
+            DEFAULT_TIME_FORMAT: 'LTS z', // example: Jan 01, 2016 15:30 UTC
+            DEFAULT_DATE_TIME_FORMAT: 'll HH[:]mm z' // example: Jan 01, 2016 15:30 UTC
+        };
+
+        var DEFAULT_LANG = 'en';
+        var DEFAULT_TIMEZONE = 'UTC';
+        /**
+         *
+         * Renders a localized date/time in a standard format.
+         *
+         * Please Note: the preferred method is via underscore in the mako
+         * templates, however this will also work.
+         *
+         * @param context
+         *
+         * - datetime: {string} The (UTC) datetime server-side string.
+         * - timezone: {string} (optional) the user set time zone preference, or 'None' if unset.
+         * This should be in a format like 'Region/Locale', e.g. 'Atlantic/Azores'. Defaults to 'UTC'.
+         * - language: {string} (optional), the user set language, an ISO 639-1 language string, defaults to 'en'
+         * - format: {string} (optional), a format constant. Defaults to 'DEFAULT_DATE_TIME_FORMAT'.
+         *
+         * @returns {string} A formatted, timezone-offset, internationalized date and time.
+         *
+         */
+        localize = function(context) {
+            // made a little longer than necessary for clarity
+            var localTime;
+            var displayTime = '';
+
+            if (isValid(context.datetime)) {
+                localTime = localizeTime(
+                    DateUtils.stringToMoment(context.datetime),
+                    context.timezone
+                );
+                displayTime = getDisplayTime(localTime, context.language, context.format);
+            }
+            return displayTime;
+        };
+
+        /**
+         * Converts a string to mutable 'moment' datetime object.
+         *
+         * The input datetimeString can be just a date, a date and time,
+         * or a complex date/time string with TZ offset. This is fairly wide-open, as
+         * many strings will satisfy this requirement.
+         * (e.g. 01-01-16 and Jan 1, 2016, and 2016-01-01 01:00:00+00:00 will work)
+         *
+         * The recommendation is to use a fully informed string, e.g. '2016-01-01 01:00:00+00:00',
+         * to limit possible date, time, and timezone ambiguities.
+         *
+         * @param {string} datetimeString. A string representation of a datetime.
+         * @returns {object} A moment.js UTC datetime object.
+         *
+        */
+        stringToMoment = function(datetimeString) {
+            var utcDateObject;
+            if (isValid(datetimeString)) {
+                utcDateObject = moment(datetimeString).utc();
+            } else {
+                utcDateObject = '';
```

Comment:
Instead of an empty string, shouldn't this return `undefined`?  As the contract of the function is that it returns an `object` and not a `string`.  Also, the docstring needs to be updated to indicate this.  (As this is an external function and you have a test for it as well.)